tetris/src/assets/world3d.tsx

```diff
@@ -0,0 +1,248 @@
+                            // right
+                            case 39:
+                                this.store.moveFigure(Axis.Z, 1);
+                                break;
+                            // left
+                            case 37:
+                                this.store.moveFigure(Axis.Z, -1);
+                                break;
+                            // space
+                            case 32:
+                                this.store.delay.current = this.store.delay.fast;
+                                break;
+                            // Num 7: rotate +X
+                            case 103:
+                                this.store.rotateFigure(Axis.X, 90);
+                                break;
+                            // Num 4: rotate -X
+                            case 100:
+                                this.store.rotateFigure(Axis.X, -90);
+                                break;
+                            // Num 8: rotate +Y
+                            case 104:
+                                this.store.rotateFigure(Axis.Y, 90);
+                                break;
+                            // Num 5: rotate -Y
+                            case 101:
+                                this.store.rotateFigure(Axis.Y, -90);
+                                break;
+                            // Num 9: rotate +Z
+                            case 105:
+                                this.store.rotateFigure(Axis.Z, 90);
+                                break;
+                            // Num 6: rotate -Z
+                            case 102:
+                                this.store.rotateFigure(Axis.Z, -90);
+                                break;
+                        }
+                        break;
+                    case BABYLON.KeyboardEventTypes.KEYUP:
+                        // console.log("KEY UP: ", kbInfo.event.keyCode);
+                        switch (kbInfo.event.keyCode) {
+                            case 32:
+                                this.store.delay.current = this.store.delay.normal;
+                                break;
+                        }
+
+                        break;
+                }
+            }
+        });
+
+        engine.runRenderLoop(() => {
+            this.scene.render();
+        });
+    }
+
+    private createCamera(canvas: HTMLCanvasElement): BABYLON.ArcRotateCamera {
+        const camera = new BABYLON.ArcRotateCamera(
+            "camera",
+            Math.PI/8,
+            7*Math.PI/16,
+            20,
+            BABYLON.Vector3.Zero(),
+            this.scene
+        );
+        camera.attachControl(canvas, true);
+        camera.inputs.remove(camera.inputs.attached.keyboard);
+        return camera;
+    }
+
+    private createLight(): BABYLON.HemisphericLight {
+        const light = new BABYLON.HemisphericLight(
+            "HemiLight",
+            new BABYLON.Vector3(2, 20, 2),
```

Comment:
please move most of numbers into constant values like `lightPosition = BABYLON.Vector3(2, 20, 2)` in this case